runFailed.js

```diff
@@ -4,18 +4,14 @@ const cypress = require('cypress');
 const fs = require('fs');
 const path = require('path');
 const appDir = process.env.INIT_CWD ? process.env.INIT_CWD : path.resolve('.');
-const cypressConfig = require(`${appDir}/cypress.config`);
 
 async function runLastFailed() {
   const noFailedTestsMessage =
     'No previous failed tests detected\nTry running tests again with cypress run';
 
   // Use the cypress environment variable failedTestDirectory for where to store failed tests
   // If not set then use the directory of the project's cypress.config where the test-results defaults to
-  const failedTestFilePath =
-    cypressConfig.env?.failedTestDirectory === undefined
-      ? `${appDir}/test-results/last-run.txt`
-      : `${cypressConfig.env.failedTestDirectory}/test-results/last-run.txt`;
+  const failedTestFilePath = `${appDir}/test-results/last-run.txt`;
```

Comment:
By using `const failedTestFilePath = ${appDir}/test-results/last-run.txt` as a solution, is your `cypress.config` file also in the same directory as `failedTestDirectory`?

- If yes, would it also be correct to say `If there are failed tests, run the following command from the directory of the project's cypress.config`?

- If no, how are your run commands/scripts detecting your config file if you are calling the `npx` script from a different directory than the `cypress.config`?


I can see this solution working if you do not set a `failedTestDirectory` and run the `npx` from the directory of the `cypress.config`, or simply set the `failedTestDirectory` to `./` when running from the directory of the `cypress.config`. 

I'm having trouble at the moment understanding how this works with `failedTestDirectory` being in a different directory than the `cypress.config`, would appreciate your insight!

Reply:
I understand your concern, but I believe the crucial point here is not the exact location of the `cypress.config` file, but rather ensuring that the path where the `.txt` file with the failed test results is generated matches the value defined in `failedTestFilePath`. To achieve this, it is necessary for the script to change the directory (`cd`) before being executed.

By using `const failedTestFilePath = ${appDir}/test-results/last-run.txt`, the file path is defined relative to the script's current directory. Therefore:

- If the `cypress.config` file is in the same directory as `failedTestDirectory`:
Then yes, it would be correct to say, "If there are failed tests, run the following command from the directory of the project's `cypress.config` file."

- If the `cypress.config` file is not in the same directory as `failedTestDirectory`:
In this case, for your commands/scripts to correctly detect your configuration file and generate the results in the desired path, you need to ensure the script changes to the appropriate directory before executing the Cypress commands. This can be done with a `cd` in the script before running the `npx` command as updated in the `README`.

In summary, the proposed solution will work if:
- You do not explicitly set `failedTestDirectory` or you set `failedTestDirectory` to `./` and run the `npx` command from the `cypress.config` directory.
- You explicitly set `failedTestDirectory` to a different path and you change your directory before executing the script.

Reply:
Thanks for the details and context, appreciate it, and I better see where you are coming from in your approach.

Could you give me an example of how you run the `npx` when `explicitly set failedTestDirectory to a different path and you change your directory before executing the script.`?

Are you also passing `--config-file` to the npm script you make so Cypress knows the path to the config?